### PR TITLE
fix(config): support multiple kubeconfig files

### LINKED ISSUE #51: Arena doesn't support multi kubeconfig file
My KUBECONFIG config as follows
```
minikube=~/.kube/config
kubenetes=~/.kube/localconfig
hongkong=~/.kube/hongkongconfig
export KUBECONFIG=$KUBECONFIG:$minikube:$kubenetes:$hongkong\
```
but when I execute`arena list`, I got the following error
```
WARN[0000] Illegal kubeconfig file: :~/.kube/config:~/.kube/localconfig:~/.kube/hongkongconfig
FATA[0000] stat :~/.kube/config:~/.kube/localconfig:~/.kube/hongkongconfig: no such file or directory
```

### PR DESCRIPTION
**Pull request by chacha923**
## Purpose of this PR

Fixes #51 by letting Arena use client-go native kubeconfig loading rules when `--config` is not explicitly set.

**Proposed changes:**

- Preserve explicit single-file kubeconfig handling for `--config`.
- Allow `KUBECONFIG=file1:file2:file3` to be parsed through client-go default precedence.
- Add unit tests for explicit kubeconfig and multi-file `KUBECONFIG` loading rules.

## Change Category

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that could affect existing functionality)
- [ ] Documentation update

### Rationale

The previous setup path copied `KUBECONFIG` into `ExplicitPath`, so a path-list such as `file1:file2:file3` was treated as one kubeconfig file and failed with `stat`. Leaving `ExplicitPath` empty for environment-based configuration lets client-go split and merge the list according to Kubernetes native behavior.

## Validation

- `go test ./pkg/apis/config`
- `go test $(go list ./... | rg -v '^github.com/kubeflow/arena/test/e2e$')`

Note: `go test ./...` still fails in the local e2e setup because `/Users/a1021500303/.kube/config` is a directory on this machine.

### CODE REVIEW

**Comment by zhuangzhewei09** [comment]
Hi @wsxiaozhang @xiaozhouX, could you please help review this PR when you have time?

This PR fixes the long-standing multi-kubeconfig issue tracked in #51. The change is intentionally scoped to kubeconfig loading behavior:

* explicit --config still uses a single kubeconfig file through ExplicitPath
* when --config is not provided, Arena now leaves ExplicitPath empty and lets client-go handle KUBECONFIG=file1:file2:file3 using the native loading rules
* unit tests are added for both explicit kubeconfig and multi-file KUBECONFIG cases

I have also marked the PR as ready for review. Thanks.

**Comment by cheyang** [comment]
@gemini-code-assist /review

File: pkg/apis/config/kubeclient.go [hunk 1]
```diff
@@ -51,33 +45,38 @@ func initKubeClient(kubeconfig string) (clientcmd.ClientConfig, *rest.Config, *k
...
+	}
+
+	if _, err = os.Stat(kubeconfig); err != nil {
+		return kubeconfig, err
 	}
-	// normalize path
-	kubeconfig = filepath.Clean(kubeconfig)
-	// change ~ to user home dir
-	kubeconfig = strings.ReplaceAll(kubeconfig, "~", currentUser.HomeDir)
 	_ = os.Setenv("KUBECONFIG", kubeconfig)
```
Comment (cheyang) [inline pkg/apis/config/kubeclient.go]: `setupExplicitKubeconfig` calls `os.Setenv("KUBECONFIG", kubeconfig)` as a side effect. This changes global process state and may surprise callers or interfere with parallel tests that rely on a stable environment. Consider whether this side effect is still required, or whether the function can return the resolved path without modifying the environment.

Comment (cheyang) [inline pkg/apis/config/kubeclient.go]: When no explicit kubeconfig is provided, the new code no longer writes to `KUBECONFIG`, which is correct because client-go reads the env directly. However, when an explicit path is given, the code still overwrites `KUBECONFIG` with that single file. Is this side effect relied on by downstream consumers, or can it be removed to keep the function pure?

File: pkg/apis/config/kubeclient_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+package config
+
+import (
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+)
+
+func TestBuildClientConfigLoadingRulesWithExplicitKubeconfig(t *testing.T) {
```
Comment (cheyang) [inline pkg/apis/config/kubeclient_test.go:25]: Consider adding a test that verifies `buildClientConfigLoadingRules` returns a non-nil error when the explicit kubeconfig path does not exist. This documents the validation behavior and guards against future regressions.

File: pkg/apis/config/kubeclient_test.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	}
+	if rules.ExplicitPath != kubeconfigPath {
+		t.Fatalf("expected ExplicitPath %q, got %q", kubeconfigPath, rules.ExplicitPath)
+	}
+	if got := os.Getenv("KUBECONFIG"); got != kubeconfigPath {
+		t.Fatalf("expected KUBECONFIG %q, got %q", kubeconfigPath, got)
+	}
+}
+
+func TestBuildClientConfigLoadingRulesWithMultipleKubeconfigEnv(t *testing.T) {
```
Comment (cheyang) [inline pkg/apis/config/kubeclient_test.go:81]: Consider adding tests for `setupExplicitKubeconfig` edge cases: a path starting with `~/`, a bare `~`, and the behavior when neither an explicit path nor `KUBECONFIG` is provided. This would increase confidence in the path-resolution logic across platforms.

File: pkg/apis/config/kubeclient.go [hunk 4]
```diff
@@ -51,33 +45,38 @@ func initKubeClient(kubeconfig string) (clientcmd.ClientConfig, *rest.Config, *k
...
 }
 
-func setupKubeconfig(kubeconfig string) (string, error) {
-	// if kubeconfig is null and env "KUBECONFIG" is not null
-	// read kubeconfig from env
+func buildClientConfigLoadingRules(kubeconfig string) (*clientcmd.ClientConfigLoadingRules, error) {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	loadingRules.DefaultClientConfig = &clientcmd.DefaultClientConfig
+
+	if kubeconfig != "" {
```
Comment (cheyang) [inline pkg/apis/config/kubeclient.go:61]: A short comment here explaining that an empty `ExplicitPath` lets client-go merge multiple files from `KUBECONFIG` would help future maintainers understand why the explicit branch is skipped.

File: pkg/apis/config/kubeclient.go (hunk 1 above)
Comment (zhuangzhewei09) [reply pkg/apis/config/kubeclient.go]: The side effect is still required by downstream `arena-kubectl` invocations, which inherit `os.Environ()`. Without synchronizing an explicit `--config` to `KUBECONFIG`, the client-go client and kubectl subprocesses could target different clusters. To keep the path-resolution helper pure, I moved the environment update out of `setupExplicitKubeconfig` and into `initKubeClient`, where the process-level initialization is explicit.

File: pkg/apis/config/kubeclient_test.go (hunk 2 above)
Comment (zhuangzhewei09) [reply pkg/apis/config/kubeclient_test.go:25]: Added TestBuildClientConfigLoadingRulesWithMissingExplicitKubeconfig to verify that a missing explicit kubeconfig returns a non-nil error and no loading rules.

File: pkg/apis/config/kubeclient.go (hunk 1 above)
Comment (zhuangzhewei09) [reply pkg/apis/config/kubeclient.go]: Thanks. I moved the environment update out of `setupExplicitKubeconfig`, so the path-resolution helper now only resolves and validates the path without modifying global process state. The required KUBECONFIG synchronization is now performed explicitly in `initKubeClient`.

File: pkg/apis/config/kubeclient.go (hunk 4 above)
Comment (zhuangzhewei09) [reply pkg/apis/config/kubeclient.go:61]: Added a comment explaining that leaving ExplicitPath unset allows client-go to merge the files listed in KUBECONFIG using its native loading precedence.

File: pkg/apis/config/kubeclient_test.go (hunk 3 above)
Comment (zhuangzhewei09) [reply pkg/apis/config/kubeclient_test.go:81]: Added coverage for `~/`, bare `~`, and the case where neither an explicit kubeconfig nor KUBECONFIG is provided. The tests also verify that path resolution no longer modifies KUBECONFIG.

**Comment by cheyang** [comment]
@gemini-code-assist /review

**Comment by cheyang** [comment]
This fix correctly delegates multi-kubeconfig handling to client-go's native loading precedence. The explicit `--kubeconfig` path still validates and expands `~` properly, and the unit tests cover the explicit, multi-file env, and missing-file cases. `go test ./pkg/apis/config/...` passes. No blockers from my review.

**Comment by zhuangzhewei09** [comment]
@xiaozhouX @wsxiaozhang PTAL~
